### PR TITLE
New Client Command: showObject <objectID> resolves #1926

### LINKED ISSUE #1926: New client command to highlight on Adventure map all objects of specified type
**Describe your proposal**
New client command to highlight on Adventure map all objects of specified type.
It should also center on first one and repeating command should center on next instance found.

A window to list all objects would be ideal :)

**Additional context**
Helpful in many situations like:
Verifying object placement on maps
Finding and Testing objects on maps
After game analysis search for Utopias or Grail or what not

### PR DESCRIPTION
**Pull request by krs0**
First working version.
image
Apparently not all objects are shown :(
The following code will not give back an artifact, or a creature or a resource, only buildings.
```
		for(const auto& objectID : context.getObjects(coordinates))
		{
			const auto* object = context.getObject(objectID);

			 if(object->typeName == searchedObject)
				 target.draw(imageHighlighted, Point(0,0));
		}
```

### CODE REVIEW

**Comment by krs0** [comment]
@IvanSavenko should this be involved in `uint8_t MapRendererDebug::checksum(IMapRendererContext & context, const int3 & coordinates)`

File: client/ClientCommandManager.cpp [hunk 1]
```diff
@@ -367,6 +367,16 @@ void ClientCommandManager::processCommand(const std::string &message, bool calle
...
 	}
+	else if(commandName == "showObject")
+	{
+		std::string objectType;
+		singleWordBuffer >> objectType;
+
+		Settings config = settings.write["session"][commandName];
+
+		config->String() = objectType;
+		printCommandMessage("Showing objects of type: " + objectType + "!", ELogLevel::INFO);
```
Comment (Nordsoft91) [inline client/ClientCommandManager.cpp:378]: Will this work?
```suggestion
		Settings config = settings.write["session"][commandName];
		singleWordBuffer >> config->String();
		printCommandMessage("Showing objects of type: " + config->String() + "!", ELogLevel::INFO);
```

Comment (krs0) [reply client/ClientCommandManager.cpp]: Yes

**Comment by krs0** [comment]
Closed. Duplicate created that solves merge submodules issues.

**Pull request closed without merging**